### PR TITLE
Client schemas

### PR DESCRIPTION
**Pull request by AbooMinister25**

### CODE REVIEW

File: client/console.py [hunk 1]
```diff
@@ -201,45 +215,56 @@ async def handle_message(self) -> str:
...
     async def send_chat_message(self) -> str:
-        response = json.dumps(
-            {
-                "type": "chat",
-                "player_name": self.main_app.name,
-                "chat_message": self.message,
-            }
+        response = ChatMessage(
+            type="chat",
+            # this function will only run when the client has registers, so self.main_app.name will never be None
```
Comment (A5rocks) [inline client/console.py:228]: Let's put an assert instead -- this way we have our invariants clearly visible and get an error if they're violated.

File: server/game_components/game_objects.py [hunk 2]
```diff
@@ -93,80 +96,140 @@ def __init__(self, _name: str, _allowed_actions: list[str]):
...
+        # TODO: Tell the player "HEY, YOU DID THIS THING"
+        if command["command"] in movement_commands:
+            valid_move = self.game.move_player(self, command["command"])
+            result = {
+                "player": self.uid,
+                "direction": command["command"],
+                "success": valid_move,
+            }
+        elif command["command"] == "flee":
+            pass  # TODO: IMPLEMENT FLEEING.
```
Comment (A5rocks) [inline server/game_components/game_objects.py]: What does fleeing mean? I'm not sure we have a game mechanic like this. (I don't think we get locked into battle like an RPG, unless we do?)

**Review by BautistaDalmasso — APPROVED** [review]
LGTM

**Comment by BautistaDalmasso** [comment]
messed up, sorry!

**Pull request closed without merging**